src/api/time/TimeAPI.js

```diff
@@ -133,24 +133,27 @@ class TimeAPI extends GlobalTimeContext {
      * @method addIndependentTimeContext
      */
     addIndependentContext(key, value, clockKey) {
-        let timeContext = this.independentContexts.get(key);
+        let timeContext = this.getIndependentContext(key);
         if (!timeContext) {
-            timeContext = new IndependentTimeContext(this, key);
-            this.independentContexts.set(key, timeContext);
+            return;
         }
 
+        //stop following upstream time context since the view has it's own
+        timeContext.resetContext();
+
         if (clockKey) {
             timeContext.clock(clockKey, value);
         } else {
             timeContext.stopClock();
             timeContext.bounds(value);
         }
 
-        this.emit('timeContext', key);
+        // Notify any nested views to update, pass in the viewKey so that particular view can skip getting an upstream context
+        this.emit('refreshContext', key);
```

Comment:
What if the TimeContext emitted `refreshContext` when `resetContext()` is called, instead of the TimeAPI (which is global). Then only the affected views would get the event? I might be misunderstanding the purpose of this event though.